cni-plugin/pkg/ipamplugin/ipam_plugin.go

```diff
@@ -474,3 +498,22 @@ func cmdDel(args *skel.CmdArgs) error {
 
 	return nil
 }
+
+// getNamespaceLabels retrieves namespace labels using Calico client's Namespaces interface
+func getNamespaceLabels(calicoClient client.Interface, namespace string) (map[string]string, error) {
+	if namespace == "" {
+		return map[string]string{}, nil
+	}
+
+	// Use Calico client's Namespaces interface
+	ns, err := calicoClient.Namespaces().Get(context.Background(), namespace, options.GetOptions{})
```

Comment:
Yep - this should just be done with a standard k8s clientset. No need to plumb it through our v3 infra.